changelog.d/1242.feature.md

```diff
@@ -0,0 +1 @@
+Add a `--fetch-missing-python` flag to all commands that accept a `--python` flag. When combined, will automatically download a standalone copy of the requested python version if it's not already available on the user's system
```

Comment:
Let's use [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).

```suggestion
Add a `--fetch-missing-python` flag to all commands that accept a `--python` flag. 

When combined, this will automatically download a standalone copy of the requested python version if it's not already available on the user's system.
```